marimo/_data/charts.py

```diff
@@ -75,81 +129,158 @@ def __init__(self, should_limit_to_10_items: bool) -> None:
+                .configure_axis(grid=False)
             )
 
-        return (
-            alt.Chart(data)
-            .mark_bar()
+        _chart_with_encodings = add_encodings(_base_chart)
+        _chart = add_common_config(_chart_with_encodings)
+        return _chart.configure_axis(grid=False)
+
+    def altair_code(self, data: str, column: str) -> str:
+        base_chart_code = dedent(f"""
+        _base_chart = (
+            alt.Chart({data})
+            .transform_aggregate(count="count()", groupby=["{column}"])
+            .transform_window(
+                rank="rank()",
+                sort=[
+                    alt.SortField("count", order="descending"),
+                    alt.SortField("{column}", order="ascending"),
+                ],
+            )
+            .transform_filter(alt.datum.rank <= 10)
+            .transform_joinaggregate(total_count="sum(count)")
+            .transform_calculate(
+                percentage="datum.count / datum.total_count"
+            )
             .encode(
-                y=alt.Y(column, type="nominal"),
-                x=alt.X("count()", type="quantitative"),
+                y=alt.Y(
+                    "{column}:N",
+                    sort="-x",
+                    axis=alt.Axis(title=None),
+                ),
+                x=alt.X("count:Q", title="{NUM_RECORDS}"),
+                tooltip=[
+                    alt.Tooltip("{column}:N"),
+                    alt.Tooltip("count:Q", format="{TOOLTIP_COUNT_FORMAT}", title="{NUM_RECORDS}"),
+                ],
             )
-            .properties(width="container")
         )
 
-    def altair_code(self, data: str, column: str) -> str:
+        _bar_chart = _base_chart.mark_bar(color="{COLOR}")
+        _text_chart = _base_chart.mark_text(align="left", dx=3).encode(
+            text=alt.Text("percentage:Q", format="{TOOLTIP_PERCENTAGE_FORMAT}")
+        )
+        """)
+
         if self.should_limit_to_10_items:
             return f"""
-            _chart = (
-                alt.Chart({data})
-                .transform_aggregate(count="count()", groupby=["{column}"])
-                .transform_window(
-                    rank="rank()",
-                    sort=[
-                        alt.SortField("count", order="descending"),
-                        alt.SortField("{column}", order="ascending"),
-                    ],
-                )
-                .transform_filter(alt.datum.rank <= 10)
-                .mark_bar()
-                .encode(
-                    y=alt.Y("{column}", type="nominal", sort="-x"),
-                    x=alt.X("count", type="quantitative"),
-                )
-                .properties(title="Top 10 {column}", width="container")
-            )
-            _chart
+            {base_chart_code}
+_chart = (
+    (_bar_chart + _text_chart)
+    .properties(title="Top 10 {column}", width="container")
+    .configure_view(stroke=None)
+    .configure_axis(grid=False)
```

Comment:
this is much nicer our UI, but for a user to copy and add to their notebook, if feels like a lot? it is nice to have the exact same view that as the sidebar, but i wonder if this can be shorter more concise